app/src/main/java/com/github/se/wanderpals/ui/screens/trip/agenda/DailyActivities.kt

```diff
@@ -0,0 +1,156 @@
+  if (dailyActivities.isEmpty()) {
+    Text(
+        text = "No activities for this date",
+        style = MaterialTheme.typography.bodyLarge,
+        color = MaterialTheme.colorScheme.secondary,
+        textAlign = TextAlign.Center,
+        modifier = Modifier.padding(16.dp).testTag("NoActivitiesMessage"))
+  } else {
+    // Display the activities for the selected date
+    LazyColumn(
+        content = {
+          items(dailyActivities.sortedBy { it.startTime }) { stop ->
+            ActivityItem(stop, onActivityItemClick)
+          }
+        })
+  }
+}
+
+/**
+ * Composable function that displays an activity item.
+ *
+ * @param stop The stop to display.
+ * @param onActivityClick Callback function triggered when the activity item is clicked,
+ */
+@Composable
+fun ActivityItem(stop: Stop, onActivityClick: (String) -> Unit) {
+  Box(modifier = Modifier.testTag(stop.stopId).fillMaxWidth()) {
+    Button(
+        onClick = { onActivityClick(stop.stopId) },
+        shape = RectangleShape,
+        modifier =
+            Modifier.height(100.dp).fillMaxWidth().testTag("activityItemButton" + stop.stopId),
+        colors = ButtonDefaults.buttonColors(containerColor = Color.Transparent)) {
+          Row(modifier = Modifier.fillMaxSize()) {
+            // Texts Column
+            Column(
+                modifier =
+                    Modifier.weight(
+                            1f) // Takes up all available space, pushing the IconButton to the right
+                        .align(Alignment.CenterVertically) // Vertically center the column content
+                        .fillMaxSize(), // Fill the available space
+                verticalArrangement = Arrangement.SpaceEvenly) {
+                  Text(
+                      text = stop.title,
+                      style = MaterialTheme.typography.bodyLarge,
+                      color = Color.Black,
+                      modifier =
+                          Modifier.wrapContentWidth(Alignment.Start)
+                              .testTag("ActivityTitle" + stop.stopId))
+                  Text(
+                      text =
+                          "${stop.startTime} - ${stop.startTime.plusMinutes(stop.duration.toLong())}",
+                      style = MaterialTheme.typography.bodyLarge,
+                      color = Color.Black,
+                      modifier =
+                          Modifier.wrapContentWidth(Alignment.Start)
+                              .testTag("ActivityTime" + stop.stopId))
+                  Text(
+                      text = stop.address,
+                      style = MaterialTheme.typography.bodyLarge,
+                      color = Color.Black,
+                      modifier =
+                          Modifier.wrapContentWidth(Alignment.Start)
+                              .testTag("ActivityAddress" + stop.stopId))
+                }
+
+            // Icon Button at the far right, centered vertically
+            IconButton(
```

Comment:
This is an iconButton since later, we would that when the user click on this icon, it brings him to the map and displaying the pin of the activity. However, we forgot to put a /*todo : implement later*/ comment. Thank you for the catch